spec/components/provider_interface/find_candidates/previously_submitted_an_application_banner_component_spec.rb

```diff
@@ -0,0 +1,94 @@
+
+    context 'when the candidate has associated applications in the current cycle and a previous cycle' do
+      let!(:last_cycle_application_form) do
+        previous_year = CycleTimetableHelper.previous_year
+        form = create(
+          :application_form,
+          :completed,
+          recruitment_cycle_year: previous_year,
+          submitted_at: CycleTimetableHelper.mid_cycle(previous_year),
+          candidate:,
+          created_at: CycleTimetableHelper.mid_cycle(previous_year),
+          updated_at: CycleTimetableHelper.mid_cycle(previous_year),
+        )
+        previous_course = create(:course, provider:)
+        course_option_2 = create(:course_option, course: previous_course)
+        create(:application_choice, course_option: course_option_2, status: 'withdrawn', application_form: form)
+
+        form
+      end
+
+      it 'renders both current cycle and previous cycle associated applications' do
+        result = render_inline(described_class.new(application_form:, current_provider_user:))
+
+        expect(result.text).to include('Important')
+        expect(result.text).to include('This candidate has submitted an application to you or a partner before')
+        expect(result.text).to include(course.name_and_code)
+        expect(result.text).to include('Received')
+        expect(result.text).to include('2023 to 2024 recruitment cycle:')
+        expect(result.text).to include(last_cycle_application_form.application_choices.first.course.name_and_code)
+        expect(result.text).to include('Withdrawn')
+      end
+    end
+
+    context 'when the candidate has associated applications only in a previous cycle' do
+      let!(:last_cycle_application_form) do
+        previous_year = CycleTimetableHelper.previous_year
+        form = create(
+          :application_form,
+          :completed,
+          recruitment_cycle_year: previous_year,
+          submitted_at: CycleTimetableHelper.mid_cycle(previous_year),
+          candidate:,
+          created_at: CycleTimetableHelper.mid_cycle(previous_year),
+          updated_at: CycleTimetableHelper.mid_cycle(previous_year),
+        )
+        previous_course = create(:course, provider:)
+        course_option_2 = create(:course_option, course: previous_course)
+        create(:application_choice, course_option: course_option_2, status: 'withdrawn', application_form: form)
+
+        form
+      end
+
+      before do
+        application_choice.destroy!
+        application_form.destroy!
+      end
+
+      it 'renders only previous cycle associated applications' do
+        result = render_inline(described_class.new(application_form: last_cycle_application_form, current_provider_user:))
+
+        expect(result.text).to include('Important')
+        expect(result.text).to include('This candidate has submitted an application to you or a partner before')
+        expect(result.text).to include('2023 to 2024 recruitment cycle:')
```

Comment:
Because you have used the dynamic 'previous year' this will break when the new cycle starts. 

Either you can freeze time to always bee mid cycle 2025, or can change this: 
```suggestion
        expect(result.text).to include("#{previous_timetable.cycle_range_name} recruitment cycle:)
```